auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java

```diff
@@ -168,4 +188,50 @@ private static Metadata toHeaders(@Nullable Map<String, List<String>> metadata)
     }
     return headers;
   }
+
+  private static volatile boolean loadedReflection;
+  private static Class<? extends Credentials> serviceAccountClass;
+  private static Constructor<? extends Credentials> jwtConstructor;
+  private static Method getScopes;
+  private static Method getClientId;
+  private static Method getClientEmail;
+  private static Method getPrivateKey;
+  private static Method getPrivateKeyId;
+
+  private static Credentials serviceAccountToJwt(Credentials creds) {
+    try {
+      if (!loadedReflection) {
+        // Safe if run multiple times, since later runs produce equivalent results.
+        synchronized (GoogleAuthLibraryCallCredentials.class) {
```

Comment:
I was under the impression modern JVMs cached reflective results, making them fast later on.